### PR TITLE
Add PHPStan types for data provider

### PR DESCRIPTION
**Pull request by ben-challis**

### CODE REVIEW

File: tests/unit/PackagesProvider/ComposerInstalledJsonPackagesProviderTest.php [hunk 1]
```diff
@@ -32,7 +32,16 @@ protected function tearDown(): void
     }
 
     /**
-     * @return iterable<array{array<mixed>, Packages, bool}>
+     * @return iterable<array{
```
Comment (marmichalski) [inline tests/unit/PackagesProvider/ComposerInstalledJsonPackagesProviderTest.php:35]: Would it make sense to define (and use?) some of those as phpstan-type in ComposerInstalledJsonPackagesProvider class and just import here? :thinking:

Comment (ben-challis) [reply tests/unit/PackagesProvider/ComposerInstalledJsonPackagesProviderTest.php:35]: Yeah absolutely

Comment (ben-challis) [reply tests/unit/PackagesProvider/ComposerInstalledJsonPackagesProviderTest.php]: Done in https://github.com/Lendable/composer-license-checker/pull/65/commits/dbb54a8332d1ec8032c78df66e97afbbf89da515

File: src/PackagesProvider/ComposerInstalledJsonPackagesProvider.php [hunk 2]
```diff
@@ -81,7 +85,7 @@ static function (mixed $package) use ($skipPackages): ?Package {
                         }
 
                         /** @var list<non-empty-string> $licenses */
-                        /** @var array{name: non-empty-string, license?: list<non-empty-string>} $package */
+                        /** @var PackageData $package */
```
Comment (marmichalski) [inline src/PackagesProvider/ComposerInstalledJsonPackagesProvider.php:88]: :heart_eyes:

**Review by marmichalski — APPROVED** [review]

**Pull request merged**